catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -796,22 +796,29 @@ QuerySources initializeSources(
           for (String id : sourceIds) {
             LOGGER.debug("Looking up source ID = {}", id);
 
-            final List<FederatedSource> fedSources =
+            List<FederatedSource> sources = new ArrayList<>();
+            sources.addAll(
                 frameworkProperties
                     .getFederatedSources()
                     .stream()
                     .filter(e -> e.getId().equals(id))
-                    .collect(Collectors.toList());
+                    .collect(Collectors.toList()));
+            sources.addAll(
```

Comment:
You could use Java's Stream.concat() method to combine the federated source and catalog store streams and then filter the resulting stream. That would reduce the duplicated code a bit and make it easier to see that we are checking both types of sources.

Reply:
Ah! I knew there was a way to do it I just couldn't figure out what the method was called.

Reply:
The concat option would definitely be cleaner here.